### PR TITLE
Kh backup local metadata

### PR DESCRIPTION
**Pull request by kurvinh**

### CODE REVIEW

**Comment by kurvinh** [comment]
Screenshot 2020-09-08 at 17 52 44

File: backup/moodle2/backup_local_metadata_plugin.class.php [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        );
+        // Connect the visible container ASAP.
+        $plugin->add_child($pluginwrapper);
+
+        // Set source to populate the data.
+        $pluginwrapper->set_source_table('local_metadata', array());
+
+        return $plugin;
+    }
+}
```
Comment (rumanod) [inline backup/moodle2/backup_local_metadata_plugin.class.php]: new line

File: backup/moodle2/restore_local_metadata_plugin.class.php [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+        if ($data->instanceid) {
+            $sql = 'SELECT * FROM {local_metadata} WHERE instanceid = ?';
+            $params = array('instanceid' => $data->instanceid);
+            if (!$DB->record_exists_sql($sql, $params)) {
+                $DB->insert_record('local_metadata', $data);
+            }
+        }
+    }
+}
```
Comment (rumanod) [inline backup/moodle2/restore_local_metadata_plugin.class.php]: new line

File: backup/moodle2/restore_local_metadata_plugin.class.php [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        $data = (object)$data;
+        $oldid = $data->id;
+        unset($data->id);
+
+        $data->instanceid = $this->task->get_courseid();
+
+        if ($data->instanceid) {
+            $sql = 'SELECT * FROM {local_metadata} WHERE instanceid = ?';
+            $params = array('instanceid' => $data->instanceid);
+            if (!$DB->record_exists_sql($sql, $params)) {
```
Comment (rumanod) [inline backup/moodle2/restore_local_metadata_plugin.class.php]: wrap in try catch so if it fails, backup restore not broken

Comment (rumanod) [reply backup/moodle2/restore_local_metadata_plugin.class.php]: All the DB queries

**Review by rumanod — CHANGES_REQUESTED** [review]

File: backup/moodle2/restore_local_metadata_plugin.class.php (hunk 3 above)
Comment (kurvinh) [reply backup/moodle2/restore_local_metadata_plugin.class.php]: 👍

**Review by rumanod — APPROVED** [review]

**Pull request merged**